src/AudioPlayerProvider.tsx

```diff
@@ -8,14 +8,33 @@ import React, {
 } from "react"
 import { Howl, HowlOptions } from "howler"
 import { initialState, reducer, Actions } from "./audioPlayerState"
-import { context } from "./context"
-import { AudioPlayerContext, AudioOptions } from "./types"
+import { playerContext, positionContext } from "./context"
+import {
+    AudioPlayerContext,
+    AudioOptions,
+    AudioPlayerPositionContext
+} from "./types"
 
 interface AudioPlayerProviderProps {
     children: React.ReactNode
     value?: AudioPlayerContext
 }
 
+const usePosition = (): AudioPlayerPositionContext => {
```

Comment:
i would argue this entire hook is unnecessary. I could see it confusing developers with the user-facing useAudioPosition hook. Let's just add the following line in its place:
`const [position, setCurrentPosition] = useState(0)`

Reply:
I see how it is useful when setting the value of the provider. If you feel its worth keeping for only that purpose then at least make sure the value is memoized:

```typescript
return useMemo(() => ({
  position,
  setPosition
}), [position, setPosition])
```

Reply:
The problem with using useState() in the AudioPlayerProvider component is that when position updates it will change AudioPlayerProvider state and thus rerender the component and all the children.

I'll add the useMemo.